src/TweetProcessor.js

```diff
@@ -194,29 +202,55 @@ class TweetProcessor {
 
 
   /**
-   * A function used to process
+   * A function used to process tweets and post them to rooms.
    *
-   * @param  {type} roomid        description
-   * @param  {type} tweets        description
-   * @param  {type} depth         description
-   * @param  {type} client = null description
-   * @return {type}               description
+   * @param  {type} rooms Rooms to post the tweets to.
+   * @param  {type} tweets Tweets to process
+   * @param  {type} opts.depth The maximum depth of the tweet chain (replies to
+   * replies) to be traversed. Set this to how deep you wish to traverse and it
+   * will be decreased when the function calls itself.
+   * @param  {type} opts.client = null The twitter authed client to use.
+   * @param  {string} opts.force_user_id Should we force one account to post for every tweet.
    */
-  process_tweets (rooms, tweets, depth, client = null) {
-    if (client == null) {
-      client = this._tclient;
+  process_tweets (rooms, tweets, opts) {
+    if (opts == null) {
+      opts = { }
     }
+    if (opts.client == null) {
+      opts.client = this._tclient;
+    }
+    if (opts.depth == null) {
+      opts.depth = DEFAULT_TWEET_DEPTH;
+    }
+    const promises = [];
     tweets.forEach( (tweet) => {
-      this._process_tweet(rooms, tweet, depth, client);
+      promises.push(this._process_tweet(rooms, tweet, opts.depth, opts));
     });
-
+    return Promise.all(promises);
   }
 
-  process_tweet (rooms, tweet, depth, client = null) {
-    if (client == null) {
-      client = this._tclient;
+  /**
+   * A function used to process a tweet and post them to rooms.
```

Comment:
> I'm not sure if doc-ing the same thing three times is the most maintainable or readable way to do things. I would probably document one of them.

Yes. I'm starting to regret this. I will document the public facing ones and have a ``see function x() for details``.